app/code/core/Mage/Sales/Model/Resource/Quote.php

```diff
@@ -156,12 +156,15 @@ public function getReservedOrderId($quote)
     /**
      * Check is order increment id use in sales/order table
      *
-     * @param string $orderIncrementId
+     * @param string|integer $orderIncrementId
+     *
      * @return boolean
      */
     public function isOrderIncrementIdUsed($orderIncrementId)
     {
         $adapter   = $this->_getReadAdapter();
+        $fields    = $adapter->describeTable($this->getMainTable());
+        $orderIncrementId = $adapter->prepareColumnValue($fields['increment_id'], $orderIncrementId);
```

Comment:
Do we need it at all here? Cant we just cast the value to string? We know which column w are binding (increment_id), so there is no need to detect the coumn type.

Reply:
There's something to say for that as well, indeed. I'll make the necessary change.